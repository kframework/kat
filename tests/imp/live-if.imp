// x |-> 1    OR    x |-> -1

int x ;
x = symbolicInt ;
if (x <= 7) {
  x = 1 ;
} else {
  x = -1 ;
}
