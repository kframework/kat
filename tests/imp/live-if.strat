16948dce0a48475505e82177d31ca328 step 2 ; ^ iftrue
1372d2417a32f37246d44d596c28192b step 2 ; which-can? (^ iftrue)
20e5fd6325cdb64621fb119372817088 step 2 ; which-can? (^ iftrue | ^ iffalse)
331ab7b225b2351b833396bdb1c5ee75 step 2 ; which-can? (^ iffalse | ^ divzero)
20e5fd6325cdb64621fb119372817088 exec-to-branch
56f33bd170a2f2956a5361ae7274ef93 compile
