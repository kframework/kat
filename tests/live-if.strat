step ; ^ iftrue
step ; which-can? (^ iftrue)
step ; which-can? (^ iftrue | ^ iffalse)
step ; which-can? (^ iffalse | ^ divzero)
exec-to-branch
