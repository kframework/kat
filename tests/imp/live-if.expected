
Strategy: step 2 ; ^ iftrue
================================================================================
<kat-imp>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        { x = 1 ; }
      </k>
      <mem>
        x |-> V0
      </mem>
    </imp>
  </harness>
</kat-imp>

Strategy: step 2 ; which-can? (^ iftrue)
================================================================================
<kat-imp>
  <s>
    #STUCK ( ) ~> #which-can ^ iftrue
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        if ( V0 <=Int 7 ) { x = 1 ; } else { x = -1 ; }
      </k>
      <mem>
        x |-> V0
      </mem>
    </imp>
  </harness>
</kat-imp>

Strategy: step 2 ; which-can? (^ iftrue | ^ iffalse)
================================================================================
<kat-imp>
  <s>
    #STUCK ( ) ~> #which-can ^ iffalse | ^ iftrue
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        if ( V0 <=Int 7 ) { x = 1 ; } else { x = -1 ; }
      </k>
      <mem>
        x |-> V0
      </mem>
    </imp>
  </harness>
</kat-imp>

Strategy: step 2 ; which-can? (^ iffalse | ^ divzero)
================================================================================
<kat-imp>
  <s>
    #STUCK ( ) ~> #which-can ^ iffalse
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        if ( V0 <=Int 7 ) { x = 1 ; } else { x = -1 ; }
      </k>
      <mem>
        x |-> V0
      </mem>
    </imp>
  </harness>
</kat-imp>

Strategy: exec-to-branch
================================================================================
<kat-imp>
  <s>
    #STUCK ( ) ~> #which-can ^ iffalse | ^ iftrue
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        if ( V0 <=Int 7 ) { x = 1 ; } else { x = -1 ; }
      </k>
      <mem>
        x |-> V0
      </mem>
    </imp>
  </harness>
</kat-imp>

Strategy: compile
================================================================================
<kat-imp>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Rules , < <imp>
        <k>
          int x , .Ids ; x = V0 ; if ( x <= 7 ) { x = 1 ; } else { x = -1 ; }
        </k>
        <mem>
          .Map
        </mem>
      </imp> --> <imp>
        <k>
          .
        </k>
        <mem>
          x |-> -1
        </mem>
      </imp> requires V0 <=Int 7 ==K false > , < <imp>
        <k>
          int x , .Ids ; x = V1 ; if ( x <= 7 ) { x = 1 ; } else { x = -1 ; }
        </k>
        <mem>
          .Map
        </mem>
      </imp> --> <imp>
        <k>
          .
        </k>
        <mem>
          x |-> 1
        </mem>
      </imp> requires V1 <=Int 7 ==K true >
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <imp>
      <k>
        .
      </k>
      <mem>
        x |-> 1
      </mem>
    </imp>
  </harness>
</kat-imp>
